### PR TITLE
Started adding plugin settings

### PR DESCRIPTION
**Pull request by zAlweNy26**
# Description

As discussed on Discord, we are introducing the plugin settings into the Cat.

Related to #370 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: core/cat/mad_hatter/mad_hatter.py [hunk 1]
```diff
@@ -137,14 +143,14 @@ def embed_tools(self):
     # Tries to load the plugin metadata from the provided plugin folder
     def get_plugin_metadata(self, plugin_folder: str):
         plugin_id = path.basename(plugin_folder)
-        plugin_json_metadata_file_name = "plugin.json"
-        plugin_json_metadata_file_path = path.join(plugin_folder, plugin_json_metadata_file_name)
-        meta = {"id": plugin_id}
+        metadata_file_name = "plugin.json"
+        metadata_file_path = path.join(plugin_folder, metadata_file_name)
+        meta = { "id": plugin_id }
```
Comment (nicola-corbellini) [inline core/cat/mad_hatter/mad_hatter.py:148]: @zAlweNy26 tuc tuc tuc, not pythonic...🐍

Comment (zAlweNy26) [reply core/cat/mad_hatter/mad_hatter.py:148]: pyth...what? @nicola-corbellini 🤒

**Comment by zAlweNy26** [comment]
@pieroit I reverted the commits about the decorator

**Comment by pieroit** [comment]
> @pieroit I reverted the commits about the decorator

Thanks. Nice experiment :)

**Pull request merged**